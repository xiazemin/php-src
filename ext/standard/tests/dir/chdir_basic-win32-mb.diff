005+ string(113) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one"
005- string(%d) "%s私はガラスを食べられますlevel_one"
009+ string(165) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one/私はガラスを食べられますlevel_two"
009- string(%d) "%s私はガラスを食べられますlevel_one%e私はガラスを食べられますlevel_two"