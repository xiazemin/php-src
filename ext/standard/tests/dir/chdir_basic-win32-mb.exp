*** Testing chdir() : basic functionality ***

-- Testing chdir() with absolute path: --
bool(true)
string(%d) "%s私はガラスを食べられますlevel_one"

-- Testing chdir() with relative paths: --
bool(true)
string(%d) "%s私はガラスを食べられますlevel_one%e私はガラスを食べられますlevel_two"
===DONE===