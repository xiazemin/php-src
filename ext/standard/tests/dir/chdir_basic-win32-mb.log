
---- EXPECTED OUTPUT
*** Testing chdir() : basic functionality ***

-- Testing chdir() with absolute path: --
bool(true)
string(%d) "%s私はガラスを食べられますlevel_one"

-- Testing chdir() with relative paths: --
bool(true)
string(%d) "%s私はガラスを食べられますlevel_one%e私はガラスを食べられますlevel_two"
===DONE===
---- ACTUAL OUTPUT
*** Testing chdir() : basic functionality ***

-- Testing chdir() with absolute path: --
bool(true)
string(113) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one"

-- Testing chdir() with relative paths: --
bool(true)
string(165) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one/私はガラスを食べられますlevel_two"
===DONE===
---- FAILED
