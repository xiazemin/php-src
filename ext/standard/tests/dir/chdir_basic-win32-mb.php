<?php
/* Prototype  : bool chdir(string $directory)
 * Description: Change the current directory 
 * Source code: ext/standard/dir.c
 */

/*
 * Test basic functionality of chdir() with absolute and relative paths
 */

echo "*** Testing chdir() : basic functionality ***\n";
$base_dir_path = dirname(__FILE__);

$level_one_dir_name = "私はガラスを食べられますlevel_one";
$level_one_dir_path = "$base_dir_path/$level_one_dir_name";

$level_two_dir_name = "私はガラスを食べられますlevel_two";
$level_two_dir_path = "$base_dir_path/$level_one_dir_name/$level_two_dir_name";

// create directories
mkdir($level_one_dir_path);
mkdir($level_two_dir_path);

echo "\n-- Testing chdir() with absolute path: --\n";
chdir($base_dir_path);
var_dump(chdir($level_one_dir_path));
var_dump(getcwd());

echo "\n-- Testing chdir() with relative paths: --\n";
var_dump(chdir($level_two_dir_name));
var_dump(getcwd());
?>
===DONE===
