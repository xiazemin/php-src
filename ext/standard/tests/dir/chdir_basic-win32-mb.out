*** Testing chdir() : basic functionality ***

-- Testing chdir() with absolute path: --
bool(true)
string(113) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one"

-- Testing chdir() with relative paths: --
bool(true)
string(165) "/Users/didi/PhpstormProjects/c/php-src/ext/standard/tests/dir/私はガラスを食べられますlevel_one/私はガラスを食べられますlevel_two"
===DONE===